subsys/settings/src/settings_nvs.c

```diff
@@ -129,12 +131,22 @@ static int settings_nvs_load(struct settings_store *cs,
 	ssize_t rc1, rc2;
 	uint16_t name_id = NVS_NAMECNT_ID;
 
+#if CONFIG_SETTINGS_NVS_NAME_CACHE
+	uint16_t cached = 0;
+
+	cf->loaded = false;
```

Comment:
This will disable the optimization if `settings_load_subtree_direct` with a handler that breaks the traversal before reaching `NVS_NAMECNT_ID` is used. It's still better than the existing behavior, so it's not a blocking comment, but just letting you know.

Reply:
Yeah, true. And also if an app loads a subtree (directly or not) several times, the cache will contain duplicated entries. The only way to make cache work correctly is to only use `settings_load()` unfortunately.

Reply:
Should we add info on that behavior to documentation?
Should we log an issue for that to be solved at some point?

Reply:
Regarding an issue I'm not certain. The cache kind of does what it is asked to do. An application should just be aware of this. I think documenting this behavior should be enough. I'll need to allocate a time to do this though.